Cores dt

krystian-hebel opened the pull request.

krystian-hebel committed:
soc/power9/chip.c: generate info about available cores for device tree

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>
Change-Id: I6b9b5355464a0637187609e5bb32d01c6175e75d

macpijan commented on src/include/cpu/power/spr.h:
```diff
@@ -14,6 +14,7 @@
 #define SPR_CIABR				0xBB
 #define SPR_DAWRX				0xBC
 #define SPR_TB					0x10C
+#define SPR_PVR					0x11F
```
One TAB too many, perhaps?

krystian-hebel replied:
Nope, GitHub has problems with rendering, it treats `+` as a separate character. Looks correct when file is opened without diff: https://github.com/3mdeb/coreboot/blob/630b8e04e4f7b3ddb3bbce8bf1c6ddc40c5205a4/src/include/cpu/power/spr.h

krystian-hebel committed:
soc/power9/chip.c: split 'memory' node in device tree

Multiple ranges can be passed either as multiple nodes or as one node
with multiple 'reg' entries, according to Devicetree Specification v0.3.
coreboot prepares one node with multiple "reg" entries (see
fit_update_memory() and functions called by it), while Skiboot expects
multiple nodes, one region each.

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>
Change-Id: Ibe1ac8458dee7b896d9b5f889a6a438ff07e24b8
